sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -658,8 +658,11 @@ static int sof_audio_probe(struct platform_device *pdev)
 		sof_audio_card_rt5682.name = card_name;
 
 		/* create speaker dai link also */
-		if (ctx->amp_type == CODEC_NONE)
+		if (ctx->amp_type == CODEC_NONE) {
 			ctx->amp_type = CODEC_RT5650;
+			ctx->ssp_amp = (sof_rt5682_quirk & SOF_SSP_PORT_AMP_MASK) >>
+					SOF_SSP_PORT_AMP_SHIFT;
+		}
```

Comment:
Can we move the `if (ctx->codec_type == CODEC_RT5650 && ctx->amp_type == CODEC_NONE)` test to `sof_intel_board_get_ctx()` as it is generic for the rt5650 codec?